kong/plugins/prometheus/handler.lua

```diff
@@ -15,9 +15,9 @@ function PrometheusHandler.init_worker()
 end
 
 
-function PrometheusHandler.log()
+function PrometheusHandler.log(self, conf)
   local message = kong.log.serialize()
-  prometheus.log(message)
+  prometheus.log(conf, message)
```

Comment:
let's move the logic like `if conf.param_collect_list then` from `exporter.lua` to this function, so you don't need to pass `conf` around. in addition to the `message` variable which is provided by PDK, let's pass an additional table that includes values extracted from our side.
```lua
function PrometheusHandler.log(self, conf)
  local message = kong.log.serialize()
  local serialized = {}
  if conf.per_consumer and message.consumer ~= nil then
      serialized.consumer = message.consumer.username
    end
  prometheus.log(message, serialized)
end
```